### PR TITLE
Initial support LLVM Intel in Windows

### PR DESCRIPTION
**Pull request by loumalouomega**
This PR updates the 'scripts/build.bat' file. The changes include setting the compiler variables 'CC' and 'CXX' based on the value of 'USE_INTEL_LLVM', and adjusting the 'CO_SIM_IO_STRICT_COMPILER' setting accordingly. When 'USE_INTEL_LLVM' is 'ON', it sets the compiler to 'icx-cl.exe' and turns off the strict compiler mode, as the Intel LLVM compiler fails with strict compiler settings. 

When 'USE_INTEL_LLVM' is 'OFF', it sets the compiler to 'cl.exe' and turns on the strict compiler mode. 

These changes are required fir for proper compilation on Windows systems when using the Intel LLVM compiler.

### CODE REVIEW

File: co_sim_io/includes/communication/pipe_communication.hpp
Comment (philbucher) [inline co_sim_io/includes/communication/pipe_communication.hpp:1]: can you please explain why these changes are required?

Seems like a wrong commit to `int` should also work in Win :D

**Comment by loumalouomega** [comment]
When I first I tried to compile with strict it was complaining about that
(not used in Windows). Then I decided to give up with the strict because I
needed to change the async io library.

**Comment by loumalouomega** [comment]
I can merge because you approved or because branch is not protected? @philbucher

**Review by philbucher — APPROVED** [review]
